### PR TITLE
[IOPLT-321] Adds all operations mapping types

### PR DESCRIPTION
**Pull request by CrisTofani**
#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/types/__tests__/boolean.test.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  it("should validate with correct values", () => {
+    const validMapping = {
+      falseString: "false",
+      mapper: "BOOLEAN_TO_STRING",
+      trueString: "true"
+    };
+
+    const result = BooleanMapping.decode(validMapping);
+
+    expect(E.isRight(result)).toBe(true);
```
Comment (AleDore) [inline src/types/__tests__/boolean.test.ts:14]:
```suggestion
    expect(E.isRight(result)).toBeTruthy();
```

File: src/types/__tests__/boolean.test.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  it("should not validate with incorrect values", () => {
+    const invalidMapping = {
+      falseString: 123,
+      mapper: "BOOLEAN_TO_STRING",
+      trueString: "true"
+    };
+
+    const result = BooleanMapping.decode(invalidMapping);
+
+    expect(E.isRight(result)).toBe(false);
```
Comment (AleDore) [inline src/types/__tests__/boolean.test.ts:26]:
```suggestion
    expect(E.isRight(result)).toBeFalsy();

File: src/types/__tests__/case.test.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  it("should validate the cases and defaultValue properties", () => {
+    const validMapping = {
+      cases: {
+        case1: "value1",
+        case2: "value2"
+      },
+      defaultValue: "default"
+    };
+    const result = SwitchCaseMapping.decode(validMapping);
+    expect(E.isRight(result)).toBe(true);
```
Comment (AleDore) [inline src/types/__tests__/case.test.ts:14]: ditto

File: src/types/__tests__/number.test.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  describe("DIVIDE_NUMBER", () => {
+    it('should validate when divider is a number and mapper is "DIVIDE_NUMBER"', () => {
+      const validData = {
+        divider: 5,
+        mapper: "DIVIDE_NUMBER"
+      };
+
+      const result = NumberMapping.decode(validData);
+
+      expect(E.isRight(result)).toBe(true);
```
Comment (AleDore) [inline src/types/__tests__/number.test.ts:14]: ditto

File: src/types/__tests__/number.test.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+    it("should not validate when divider is not a number", () => {
+      const invalidData = {
+        divider: "invalid",
+        mapper: "DIVIDE_NUMBER"
+      };
+
+      const result = NumberMapping.decode(invalidData);
+
+      expect(E.isRight(result)).toBe(false);
```
Comment (AleDore) [inline src/types/__tests__/number.test.ts:25]: ditto

File: src/types/__tests__/number.test.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+    it('should not validate when mapper is not "DIVIDE_NUMBER"', () => {
+      const invalidData = {
+        divider: 5,
+        mapper: "INVALID_MAPPER"
+      };
+
+      const result = NumberMapping.decode(invalidData);
+
+      expect(E.isRight(result)).toBe(false);
```
Comment (AleDore) [inline src/types/__tests__/number.test.ts:36]: ditto

File: src/types/__tests__/number.test.ts [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+    it('should not validate when mapper is "DIVIDE_NUMBER" and decimals attribute is evaluated', () => {
+      const invalidData = {
+        decimals: 2,
+        mapper: "DIVIDE_NUMBER"
+      };
+
+      const result = NumberMapping.decode(invalidData);
+
+      expect(E.isRight(result)).toBe(false);
```
Comment (AleDore) [inline src/types/__tests__/number.test.ts:47]: ditto

File: src/types/__tests__/number.test.ts [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  describe("ROUND_NUMBER", () => {
+    it('should validate when decimals is a number and mapper is "ROUND_NUMBER"', () => {
+      const validData = {
+        decimals: 5,
+        mapper: "ROUND_NUMBER"
+      };
+
+      const result = NumberMapping.decode(validData);
+
+      expect(E.isRight(result)).toBe(true);
```
Comment (AleDore) [inline src/types/__tests__/number.test.ts:60]: ditto

File: src/types/__tests__/number.test.ts [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+    it("should not validate when decimals is not a number", () => {
+      const invalidData = {
+        decimals: "5",
+        mapper: "ROUND_NUMBER"
+      };
+
+      const result = NumberMapping.decode(invalidData);
+
+      expect(E.isRight(result)).toBe(false);
```
Comment (AleDore) [inline src/types/__tests__/number.test.ts:71]: ditto

File: src/types/__tests__/number.test.ts [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+    it('should not validate when mapper is not "ROUND_NUMBER"', () => {
+      const invalidData = {
+        decimals: 5,
+        mapper: "INVALID"
+      };
+
+      const result = NumberMapping.decode(invalidData);
+
+      expect(E.isRight(result)).toBe(false);
```
Comment (AleDore) [inline src/types/__tests__/number.test.ts:82]: ditto

File: src/types/__tests__/number.test.ts [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+    it('should not validate when mapper is "ROUND_NUMBER" and divider attribute is evaluated', () => {
+      const invalidData = {
+        divider: 2,
+        mapper: "ROUND_NUMBER"
+      };
+
+      const result = NumberMapping.decode(invalidData);
+
+      expect(E.isRight(result)).toBe(false);
```
Comment (AleDore) [inline src/types/__tests__/number.test.ts:93]: ditto

File: src/types/__tests__/case.test.ts (hunk 3 above)
Comment (CrisTofani) [reply src/types/__tests__/case.test.ts:14]:
```suggestion
    expect(E.isRight(result)).toBeTry;
```

File: src/types/__tests__/number.test.ts (hunk 4 above)
Comment (CrisTofani) [reply src/types/__tests__/number.test.ts:14]:
```suggestion
      expect(E.isRight(result)).toBeTruthy();
```

File: src/types/__tests__/number.test.ts (hunk 5 above)
Comment (CrisTofani) [reply src/types/__tests__/number.test.ts:25]:
```suggestion
      expect(E.isRight(result)).toBeFalsy();
```

File: src/types/__tests__/number.test.ts (hunk 6 above)
Comment (CrisTofani) [reply src/types/__tests__/number.test.ts:36]:
```suggestion
      expect(E.isRight(result)).toBeFalsy();
```

File: src/types/__tests__/number.test.ts (hunk 7 above)
Comment (CrisTofani) [reply src/types/__tests__/number.test.ts:47]:
```suggestion
      expect(E.isRight(result)).toBeFalsy();
```

File: src/types/__tests__/number.test.ts (hunk 10 above)
Comment (CrisTofani) [reply src/types/__tests__/number.test.ts:82]:
```suggestion
      expect(E.isRight(result)).toBeFalsy();
```

**Review by AleDore — APPROVED** [review]
lgtm

**Pull request merged**